### PR TITLE
Board Checker Functionality

### PR DESCRIPTION
**Pull request by ajongaro**
Organized and categorized tests
Added 'possible wins' method to compile all the rows/columns/diags into a single array
Added winner? method to see if anyone has won.
  - we might consider moving this to another class later potentially...

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -18,10 +19,50 @@ def create_layout
...
+    diag_9,
+    diag_10,
+    diag_11,
+    diag_12
+  ]
+  end
+
+  def winner?
+    possible_wins.each do |line|
+     return true if line.include?("XXXX") || line.include?("OOOO")
```
Comment (BRohrig) [inline lib/board.rb:55]: my thought was to add an argument to the winner? method that calls whichever player token just finished a turn to slightly reduce the amount of checking done, but that should be an easy update

File: lib/board.rb [hunk 2]
```diff
@@ -18,10 +19,50 @@ def create_layout
...
+    end
+    false # if no connect 4's are found
```
Comment (BRohrig) [inline lib/board.rb:57]: probably not in this method, but we need to make an evaluation for a tie game condition if all board spaces are filled

File: spec/board_spec.rb [hunk 3]
```diff
@@ -190,8 +198,25 @@
...
+  end
+
+  describe '#winner?' do
+    it 'returns false with no pieces played' do
+      board.create_layout
+      
+      expect(board.winner?).to be false
+    end
 
+    it 'returns true with 4 pieces in a _row_' do
```
Comment (BRohrig) [inline spec/board_spec.rb:210]: maybe change the "row" here to "in a line", since we're using row to describing horizontal as "row"

**Review by BRohrig — APPROVED** [review]

File: spec/board_spec.rb (hunk 3 above)
Comment (ajongaro) [reply spec/board_spec.rb:210]: I'm confused about this, we'll cover it later. I thought this was checking for a horizontal row.

**Pull request merged**